gui/mozregui/report_delegate.py

```diff
@@ -0,0 +1,51 @@
+from PyQt4.QtGui import QStyledItemDelegate, QStyleOptionProgressBarV2, \
+    QApplication, QStyle
+from PyQt4.QtCore import Qt, QRect, pyqtSlot as Slot
+
+
+class ReportItemDelegate(QStyledItemDelegate):
+    def __init__(self, parent=None, *args):
+        QStyledItemDelegate.__init__(self, parent, *args)
+        self.progress = 0
+        self.lastIndex = None
+
+    @Slot(object, int, int)
+    def show_dl_progress(self, dl, current, total):
+        self.progress = (current * 100) / total
+        if self.lastIndex:
+            self.lastIndex.model().dataChanged.emit(
+                self.lastIndex, self.lastIndex)
+
+    def paint(self, painter, option, index):
+        # Write status
+        QStyledItemDelegate.paint(self, painter, option, index)
+
+        if not index.row() == (index.model().rowCount() - 1):
```

Comment:
I feel like we should ask the reportitem instance here if it is downloading. Maybe the reportitem could also store the percent information. Or it is possible to store this information for now on the ReportModel directly, as we are downloading only one at a time.